### PR TITLE
Fix errors in dialect stubs

### PR DESCRIPTION
**Pull request by bryanforbes**

### CODE REVIEW

File: sqlalchemy-stubs/dialects/mssql/__init__.pyi [hunk 1]
```diff
@@ -1,38 +1,33 @@
...
+    BIT as BIT,
+    REAL as REAL,
+    IMAGE as IMAGE,
+    TIMESTAMP as TIMESTAMP,
+    ROWVERSION as ROWVERSION,
+    MONEY as MONEY,
+    SMALLMONEY as SMALLMONEY,
+    UNIQUEIDENTIFIER as UNIQUEIDENTIFIER,
+    SQL_VARIANT as SQL_VARIANT,
+    XML as XML
```
Comment (ilevkivskyi) [inline sqlalchemy-stubs/dialects/mssql/__init__.pyi]: It is always good to minimize the diff to simplify comparison. For example here it looks like imports are sorted differently in your branch.

File: sqlalchemy-stubs/dialects/mssql/base.pyi [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+class _DateTimeBase(object): ...
+
+class SMALLDATETIME(_DateTimeBase, sqltypes.DateTime):
+    __visit_name__: str = ...
+
+class DATETIME2(_DateTimeBase, sqltypes.DateTime):
+    __visit_name__: str = ...
+
+class DATETIMEOFFSET(sqltypes.TypeEngine):
```
Comment (ilevkivskyi) [inline sqlalchemy-stubs/dialects/mssql/base.pyi]: I prefer to always have explicit type arguments for generics (even if it is `Any`). For example, here it is better to write `TypeEngine[Any]`.

File: sqlalchemy-stubs/dialects/postgresql/array.pyi [hunk 3]
```diff
@@ -0,0 +1,18 @@
+from typing import Any as AnyType, Optional, TypeVar, Type, Callable, overload
+from ...sql import expression
+from ...sql.type_api import TypeEngine
+from ... import types as sqltypes
```
Comment (ilevkivskyi) [inline sqlalchemy-stubs/dialects/postgresql/array.pyi]: Note that all imports of the form `from x import y as z` are treated as public by type checkers. I they are actually "private" (e.g. needed only for annotations), then you should give them a name starting with underscore, for example
```python
from ... import types as _sqltypes
```

**Comment by bryanforbes** [comment]
I believe I followed the typeshed style guide, but I could be mistaken. Are there other violations aside from those you've pointed out?

File: sqlalchemy-stubs/dialects/postgresql/array.pyi (hunk 3 above)
Comment (bryanforbes) [reply sqlalchemy-stubs/dialects/postgresql/array.pyi]: Is this still the case even if the library uses `from x import y as z`? For instance, [postgres/array.py](https://bitbucket.org/zzzeek/sqlalchemy/src/6e958acf49a606b176e635a216ace95813ffa6a6/lib/sqlalchemy/dialects/postgresql/array.py?at=rel_1_2&fileviewer=file-view-default#array.py-11) import this line as I have it in the typings.

File: sqlalchemy-stubs/dialects/mssql/base.pyi (hunk 2 above)
Comment (bryanforbes) [reply sqlalchemy-stubs/dialects/mssql/base.pyi]: I'll update those. Thanks!

File: sqlalchemy-stubs/dialects/mssql/__init__.pyi (hunk 1 above)
Comment (ilevkivskyi) [reply sqlalchemy-stubs/dialects/mssql/__init__.pyi]: It is actually easier to review. For example I can easily spot whether something is deleted.  (Sorry if this sounds a bit paranoid.)

File: sqlalchemy-stubs/dialects/postgresql/array.pyi (hunk 3 above)
Comment (ilevkivskyi) [reply sqlalchemy-stubs/dialects/postgresql/array.pyi]: Yes, this is still true even if name changes (at least for mypy). The general rule is that names not "officially" advertised as part of public API stay private. For example, `from dataclasses import sys` should be an error even though this line will work at runtime.

**Comment by ilevkivskyi** [comment]
> I believe I followed the typeshed style guide, but I could be mistaken. Are there other violations aside from those you've pointed out?

No, I posted the link just in case, it is great you already use the guide.

**Comment by bryanforbes** [comment]
Thank you for the feedback. I've updated the pull request based on it.

**Pull request merged**